src/monster.h

```diff
@@ -218,7 +231,17 @@ class Monster final : public Creature
 	uint32_t getDamageImmunities() const override { return mType->info.damageImmunities; }
 	uint32_t getConditionImmunities() const override { return mType->info.conditionImmunities; }
 	void getPathSearchParams(const Creature* creature, FindPathParams& fpp) const override;
-	bool useCacheMap() const override { return !randomStepping; }
+
+	// map cache
+	bool mapLoaded = false;
+	bool walkabilityCache[walkCacheHeight][walkCacheWidth] = {{false}};
```

Comment:
Note this will only set the first value of the first nested array to false, not every entry. You probably just want

```suggestion
	bool walkabilityCache[walkCacheHeight][walkCacheWidth] = {};
```

Or even a [`bitset`](https://en.cppreference.com/w/cpp/utility/bitset)

Reply:
Right, I'll try using bitset later